src/python/dxpy/scripts/dx.py

```diff
@@ -4397,6 +4409,17 @@ def register_parser(parser, subparsers_action=None, categories=('other', ), add_
 parser_list_stages.set_defaults(func=workflow_cli.list_stages)
 register_parser(parser_list_stages, subparsers_action=subparsers_list, categories='workflow')
 
+parser_list_files = subparsers_list.add_parser('files', help='List the data files associated with a database',
+                                                description='List the data files associated with a database',
+                                                parents=[env_args],
+                                                prog='dx list files')
+parser_list_files.add_argument('database', help='ID of the database.')
+parser_list_files.add_argument('--filename', help='Directory from which to start search. This will typically match the name of the table whose files are of interest. If omitted, the search starts at the root location for the database.')
+parser_list_files.add_argument("--recursive", default=False, help='Look for files recursively down the directory structure. Otherwise, by default, only look on one level.', action='store_true')
+parser_list_files.add_argument("--timeout", default=120, help='Number of seconds to wait before aborting the request. If omitted, default timeout is 120 seconds.', type=int)
```

Comment:
How do timeouts work?

Reply:
@wkoetke i am a bit hesitant to add timeout here. don't think users will care to add that. we just should have something in the api server that does not take more than a minute to list files..

Reply:
@arunbhat, see my comment in the back end PR. Might be able to move the listing to the client

Reply:
I added 'timeout' because it was requested during one of our standup sessions. Note some other dx commands also have 'timeout' so its even consistent. There is already an implicit timeout for our meta data fetching routes (getStatusList, renameFile, deleteFile) in the form of "apiserver.tip.metaMaxDuration" property. Default 600 seconds I think. So that also acts as a ceiling to showFiles. The 'timeout' cmd line arg allows the user to override this explicitly with a LOWER value if they aren't willing to wait that long, since it's a synchronous user command. Not sure why this is so controversial but I'll remove it if people continue to be troubled by it. Seems simple and straightforward in my opinion.